addons/website_event_exhibitor/views/event_templates_sponsor.xml

```diff
@@ -32,4 +24,16 @@
     </xpath>
 </template>
 
+<!-- Common template for sponsor images and 'Unpublished' badge -->
+<template id="event_sponsor_thumb_details">
+    <div class="h-100 shadow-sm p-2">
+        <span t-field="sponsor.image_128"
+            t-options='{"widget": "image", "class": "img img-fluid"}'/>
+        <span t-if="sponsor.sponsor_type_id.display_ribbon_style and sponsor.sponsor_type_id.display_ribbon_style != 'no_ribbon'"
+            t-field="sponsor.sponsor_type_id" t-attf-class="o_ribbon o_ribbon_right ribbon_#{sponsor.sponsor_type_id.display_ribbon_style}"/>
+    </div>
+    <span t-if="isSponsorUnpublished"
```

Comment:
Seems it could be computed based on sponsor. It is not given in second call of this template, is it intended ?

Reply:
Right, used the same variable for both calls of template.